Sources/Plasma/NucleusLib/pnAsyncCoreExe/pnAceSocket.cpp

```diff
@@ -471,18 +472,31 @@ static bool SocketQueueAsyncWrite(AsyncSocket conn, const void* data, unsigned b
         op->fNotify.asyncId = /* TODO */nullptr;
         op->fNotify.buffer = membuf + sizeof(WriteOperation);
         op->fNotify.bytes = bytes;
-        op->fNotify.bytesProcessed = bytes;
+        op->fNotify.bytesProcessed = 0;
         memcpy(op->fNotify.buffer, data, bytes);
 
         PerfAddCounter(kAsyncPerfSocketBytesWaitQueued, bytes);
     }
 
     std::vector<asio::const_buffer> allWrites;
     allWrites.reserve(conn->fWriteOps.size());
-    for (const WriteOperation* op : conn->fWriteOps)
+    for (WriteOperation* op : conn->fWriteOps) {
         allWrites.emplace_back(op->AsBuffer());
-    conn->fSock.async_write_some(allWrites, [conn](const asio::error_code& err, size_t bytes) {
-        //...
+        op->fNotify.bytesCommitted = op->fNotify.bytes;
+    }
+    async_write(conn->fSock, allWrites, [conn, allWrites](const asio::error_code& err, size_t bytes) {

```

Comment:
Do `conn` and `allWrites` need to be copied by value, or can we use references or move them into the lamdba? By move, I mean use `std::,move()` to move the contents in the capture.

Reply:
allWrites looks like more dead code. conn could simply be passed by reference using & in since the lifetime of conn should not exceed the lambda. At least - if it does exceed the lambdas lifetime - something else is very wrong.

Reply:
I committed a change to make conn pass by reference, but conn is actually a pointer type so that may have been kind of silly.

Reply:
Yeah, if conn is a pointer, copying by value is fine. I was primarily worried about copying the vector `allWrites`, which could be expensive (eg require an extra allocation).

Reply:
I've removed the commit that pass `conn` by reference. `allWrites` is no longer passed through at all.